### PR TITLE
MNIST offline mode with MXNet/PyTorch/TensorFlow

### PR DESCRIPTION
**Pull request by hsl89**
*Issue #, if available:*

*Description of changes:*

1. Added notebooks in tensorflow / pytorch/ mxnet for training / deployment using sagemaker python sdk v2
2. Used latest version of all deep learning frameworks
3. Download MNIST data from S3 bucket 
4. Removed user interactive inputs

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by aaronmarkham — DISMISSED** [review]

**Review by aaronmarkham — CHANGES_REQUESTED** [review]
Let's try to get these to pass CI.

File: frameworks/pytorch/get_started_mnist_train.ipynb [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+    "it with the following parameters to set up the environment:\n",
+    "\n",
+    "- entry_point: A user defined python file to be used by the training container as the \n",
+    "instructions for training. We further discuss this file in the next subsection.\n",
+    "\n",
+    "- role: An IAM role to make AWS service requests\n",
+    "\n",
+    "- instance_type: The type of SageMaker instance to run your training script. \n",
+    "Set it to `local` if you want to run the training job on \n",
+    "the SageMaker instance you are using to run this notebook\n",
```
Comment (aaronmarkham) [inline frameworks/pytorch/get_started_mnist_train.ipynb:54]: Add comment that local mode is not supported in Studio.

File: frameworks/tensorflow/get_started_mnist_train.ipynb [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+    "it with the following parameters to set up the environment:\n",
+    "\n",
+    "- entry_point: A user defined python file to be used by the training container as the \n",
+    "instructions for training. We will further discuss this file in the next subsection\n",
+    "\n",
+    "- role: An IAM role to make AWS service requests\n",
+    "\n",
+    "- instance_type: The type of SageMaker instance to run your training script. \n",
+    "Set it to `local` if you want to run the training job on \n",
+    "the SageMaker instance you are using to run this notebook\n",
```
Comment (aaronmarkham) [inline frameworks/tensorflow/get_started_mnist_train.ipynb:52]: Add note about not supported on Studio.

File: frameworks/mxnet/get_started_mnist_train.ipynb [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+    "it with the following parameters to set up the environment:\n",
+    "\n",
+    "- entry_point: A user defined python file to be used by the training container as the \n",
+    "instructions for training. We further discuss this file in the next subsection.\n",
+    "\n",
+    "- role: An IAM role to make AWS service requests\n",
+    "\n",
+    "- instance_type: The type of SageMaker instance to run your training script. \n",
+    "Set it to `local` if you want to run the training job on \n",
+    "the SageMaker instance you are using to run this notebook\n",
```
Comment (aaronmarkham) [inline frameworks/mxnet/get_started_mnist_train.ipynb:54]: Add note about not supported on Studio

File: frameworks/mxnet/get_started_mnist_train.ipynb [hunk 4]
```diff
@@ -0,0 +1,411 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pygmentize code/test_train.py"
+   ]
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "Python 3",
```
Comment (aaronmarkham) [inline frameworks/mxnet/get_started_mnist_train.ipynb:391]: Run in studio and set to mxnet cpu optimized.

File: frameworks/pytorch/get_started_mnist_train.ipynb [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pygmentize code/test_train.py"
+   ]
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "Python 3",
```
Comment (aaronmarkham) [inline frameworks/pytorch/get_started_mnist_train.ipynb:388]: Run in studio and set to Pytorch cpu optimized

File: frameworks/tensorflow/get_started_mnist_train.ipynb [hunk 6]
```diff
@@ -0,0 +1,418 @@
...
+   "metadata": {},
+   "source": [
+    "In [the next notebook](get_started_mnist_deploy.ipynb) you will see how to deploy your \n",
+    "trained model artifacts to a SageMaker endpoint. "
+   ]
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "Python 3",
```
Comment (aaronmarkham) [inline frameworks/tensorflow/get_started_mnist_train.ipynb:398]: Run in studio and set to TF2 cpu optimized

File: frameworks/mxnet/get_started_mnist_deploy.ipynb [hunk 7]
```diff
@@ -0,0 +1,501 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Deploy a Trained Model\n",
```
Comment (aaronmarkham) [inline frameworks/mxnet/get_started_mnist_deploy.ipynb:7]:
```suggestion
    "# Deploy a Trained MXNet Model\n",
```

File: frameworks/pytorch/get_started_mnist_deploy.ipynb [hunk 8]
```diff
@@ -0,0 +1,526 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Deploy a Trained Model\n",
```
Comment (aaronmarkham) [inline frameworks/pytorch/get_started_mnist_deploy.ipynb:7]:
```suggestion
    "# Deploy a Trained PyTorch Model\n",
```

File: frameworks/tensorflow/get_started_mnist_deploy.ipynb [hunk 9]
```diff
@@ -0,0 +1,296 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Deploy a Trained Model\n",
```
Comment (aaronmarkham) [inline frameworks/tensorflow/get_started_mnist_deploy.ipynb:7]:
```suggestion
    "# Deploy a Trained TensorFlow 2 Model\n",
```

File: sagemaker-python-sdk/tensorflow_serving_container/tensorflow_serving_container.ipynb [hunk 10]
```diff
@@ -317,7 +317,7 @@
  ],
  "metadata": {
   "kernelspec": {
-   "display_name": "conda_tensorflow_p36",
+   "display_name": "Environment (conda_tensorflow_p36)",
```
Comment (aaronmarkham) [inline sagemaker-python-sdk/tensorflow_serving_container/tensorflow_serving_container.ipynb:320]: change unrelated to this PR, so remove

File: .gitignore [hunk 11]
```diff
@@ -3,4 +3,10 @@
 **/__pycache__
 **/.aws-sam
 .DS_Store
+<<<<<<< HEAD
+
+*.swp
+*.swo
+=======
 **/_build
+>>>>>>> b0ebb0b58bc0a62c92abb3fc2c618ba98ad946c2
```
Comment (aaronmarkham) [inline .gitignore:12]: I'd remove this file from the PR

File: frameworks/tensorflow/.gitignore [hunk 12]
```diff
@@ -0,0 +1,2 @@
+temp.ipynb
```
Comment (aaronmarkham) [inline frameworks/tensorflow/.gitignore:1]: Is this required? I'd remove it, if not.

**Review by aaronmarkham — CHANGES_REQUESTED** [review]

File: frameworks/pytorch/get_started_mnist_train.ipynb (hunk 1 above)
Comment (hsl89) [reply frameworks/pytorch/get_started_mnist_train.ipynb:54]: A comment is added before calling `.deploy`

File: frameworks/tensorflow/get_started_mnist_deploy.ipynb (hunk 9 above)
Comment (hsl89) [reply frameworks/tensorflow/get_started_mnist_deploy.ipynb:7]: done

File: frameworks/pytorch/get_started_mnist_deploy.ipynb (hunk 8 above)
Comment (hsl89) [reply frameworks/pytorch/get_started_mnist_deploy.ipynb:7]: donw

**Review by aaronmarkham — APPROVED** [review]
Congrats! It passed CI!!!

**Pull request merged**